### PR TITLE
feat: enable hermes on android

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [MOPLAT-732] 

### Description

Re-enables hermes on android to investigate https://artsyproduct.atlassian.net/browse/MOPLAT-732

Turns out updating hermes-engine to version 0.10.0 fixes the issue that we were having, we def need to further QA it but so far it looks good! 

Found the mapping of react-native versions to hermes versions [here](https://github.com/facebook/hermes/releases)

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

- enable hermes on android - gkartalis

#### Dev changes

- 

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[MOPLAT-732]: https://artsyproduct.atlassian.net/browse/MOPLAT-732?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: android/app/proguard-rules.pro [hunk 1]
```diff
@@ -8,3 +8,7 @@
 #   http://developer.android.com/guide/developing/tools/proguard.html
 
 # Add any project specific keep options here:
+
+# needed for hermes
+-keep class com.facebook.hermes.unicode.** { *; }
+-keep class com.facebook.jni.** { *; }
```
Comment (gkartalis) [inline android/app/proguard-rules.pro:14]:
```suggestion
-keep class com.facebook.jni.** { *; }

```

**Comment by gkartalis** [comment]
exciting! Seems like updating hermes-engine to version 10 fixes the issue! 🥇 found [here](https://github.com/facebook/hermes/releases) a "mapping" of rn-versions and hermes-engine versions

**Review by MounirDhahri — APPROVED** [review]
LGTM - thanks for getting back to this. @artsy/mobile-platform how about we create a beta and ask our 3rd party QA vendor to review this over the weekend?

**Comment by gkartalis** [comment]
@MounirDhahri Almost there, trying to figure out if `"hermes-engine": "0.10.0"` is stable though 🤔 there are some comments on [this issue](https://github.com/facebook/hermes/issues/649#issuecomment-1073762024) about some hermes issues with incorrect bytecode still so not sure if we should first upgrade react-native and then do hermes

**Comment by gkartalis** [comment]
Closing this due to https://github.com/artsy/eigen/pull/8658

**Pull request closed without merging**